Mathlib/Data/Real/Basic.lean

```diff
@@ -590,6 +590,16 @@ lemma mul_add_one_le_add_one_pow {a : ℝ} (ha : 0 ≤ a) (b : ℕ) : a * b + 1
       exact (pow_le_pow_left zero_le_one (by simpa using ha'.le) b).trans' (by simp)
     · exact hb ha'
 
+/-- If `u v : ℕ → ℝ` are nonnegative and bounded above, then `u * v` is bounded above. -/
+theorem bddAbove_range_mul {u v : ℕ → ℝ} (hu : BddAbove (Set.range u)) (hu0 : 0 ≤ u)
+    (hv : BddAbove (Set.range v)) (hv0 : 0 ≤ v) : BddAbove (Set.range (u * v)) := by
```

Comment:
This is true in much greater generality. The domain can be arbitrary, and I think the codomain only needs to have `Zero`, `Mul` and `PosMulMono` and `MulPosMono`, or something to that effect (e.g., `OrderedSemiring`), because you simply need to apply [mul_le_mul_of_nonneg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/GroupWithZero/Unbundled.html#mul_le_mul_of_nonneg).

If you need things like this, I suggest duplicating the API around [BddAbove.mul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/Pointwise/Bounds.html#BddAbove.mul) with these alternative assumptions.